src/meteodatalab/operators/vertical_extrapolation.py

```diff
@@ -0,0 +1,114 @@
+"""Vertical extrapolation operators."""
+
+# Third-party
+import numpy as np
+import xarray as xr
+
+# Local
+from .. import physical_constants as pc
+
+LAPSE_RATE = 0.0065  # K m^-1
+H1 = 2000.0
+H2 = 2500.0
+T1 = 298.0
+
+
+def extrapolate_temperature_sfc2p(
+    t_sfc: xr.DataArray,
+    z_sfc: xr.DataArray,
+    p_sfc: xr.DataArray,
+    p_target: float,
+) -> xr.DataArray:
+    """Extrapolate temperature to a target pressure level.
+
+    Implements the algorithm described in [1]_. The algorithm extrapolates
+    temperature from the surface to a target pressure level using a
+    polynomial expression of a dimensionless variable y, which is a function of
+    the surface temperature, surface pressure, and geopotential. It assumes
+    a constant lapse rate of 0.0065 K m^-1.
+
+    Parameters
+    ----------
+    t_sfc : xr.DataArray
+        Surface temperature [K].
+    z_sfc : xr.DataArray
+        Surface geopotential.
+    p_sfc : xr.DataArray
+        Surface pressure.
+    p_target : float
+        Target pressure level.
+
+    Returns
+    -------
+    xr.DataArray
+        Extrapolated temperature at the target pressure level.
+
+    References
+    ----------
+    .. [1] https://www.umr-cnrm.fr/gmapdoc/IMG/pdf/ykfpos46t1r1.pdf
```

Comment:
This URL does not look very durable. Is there any other source for the reference?

Reply:
I found this other link: https://www.ecmwf.int/sites/default/files/elibrary/2003/77039-ifs-documentation-cy25r1-part-vi-technical-and-computational-procedures_1.pdf , it might be slightly better in that regard. What do you think?

Unfortunately the origin of this algorithm is unclear..